doc(website): Slack plugin blog post (#5807)
cloudquery/cloudquery: website/pages/blog/introducing-the-slack-source-plugin.md

---
title: Introducing the Slack Source Plugin
tag: announcement
date: 2022/12/20
description: >-
  CloudQuery is an open-source, extensible framework that gives you a
  single-pane-of-glass to your cloud-infrastructure using SQL. Today, we are
  happy to announce the release of the Slack source plugin.
author: hermanschaaf
---

import { BlogHeader } from "../../components/BlogHeader"

<BlogHeader/>

> CloudQuery is an open-source, extendable framework that gives you a single-pane-of-glass to your cloud-infrastructure using SQL. Today, we are happy to announce the release of the Slack source plugin for CloudQuery.

Reviewer: ```suggestion
CloudQuery is an open-source, extendable framework that gives you a single-pane-of-glass to your cloud-infrastructure. Today, we are happy to announce the release of the Slack source plugin for CloudQuery.
```

The SQL part was removed in some of the recent posts (vercel)

> CloudQuery is an open-source, extendable framework that gives you a single-pane-of-glass to your cloud-infrastructure using SQL. Today, we are happy to announce the release of the Slack source plugin for CloudQuery.

Reviewer 2: Can we use the same tagline as on the website?

Author: @yevgenypats "CloudQuery is an open source high performance data integration platform designed for security and infrastructure teams." ?

Author: Updated

> CloudQuery is an open-source, extendable framework that gives you a single-pane-of-glass to your cloud-infrastructure using SQL. Today, we are happy to announce the release of the Slack source plugin for CloudQuery.

Author: ```suggestion
CloudQuery is an open source high performance data integration platform designed for security and infrastructure teams. Today, we are happy to announce the release of the Slack source plugin for CloudQuery.
```

Slack is a popular communication tool used by many organizations. With the CloudQuery Slack source plugin, you can now load Slack workspace data into Postgres, Snowflake, BigQuery, SQLite, or any other destination supported by CloudQuery. 

As part of its initial release, the Slack plugin supports pulling data for the following Slack resources:

>  - users into the [slack_users](https://github.com/cloudquery/cloudquery/blob/main/plugins/source/slack/docs/tables/slack_users.md) and [slack_user_presences](https://github.com/cloudquery/cloudquery/blob/main/plugins/source/slack/docs/tables/slack_user_presences.md) tables,

Reviewer 3: ```suggestion
 - users into the [`slack_users`](https://github.com/cloudquery/cloudquery/blob/main/plugins/source/slack/docs/tables/slack_users.md) and [slack_user_presences](https://github.com/cloudquery/cloudquery/blob/main/plugins/source/slack/docs/tables/slack_user_presences.md) tables,
```

Quoting the link text should fix these

>  - users into the [`slack_users`](https://github.com/cloudquery/cloudquery/blob/main/plugins/source/slack/docs/tables/slack_users.md) and [slack_user_presences](https://github.com/cloudquery/cloudquery/blob/main/plugins/source/slack/docs/tables/slack_user_presences.md) tables,

Author: ```suggestion
 - users into the [`slack_users`](https://github.com/cloudquery/cloudquery/blob/main/plugins/source/slack/docs/tables/slack_users.md) and [`slack_user_presences`](https://github.com/cloudquery/cloudquery/blob/main/plugins/source/slack/docs/tables/slack_user_presences.md) tables,
```

 - teams into the [`slack_teams`](https://github.com/cloudquery/cloudquery/blob/main/plugins/source/slack/docs/tables/slack_teams.md) table,
 - channels into the [`slack_conversations`](https://github.com/cloudquery/cloudquery/blob/main/plugins/source/slack/docs/tables/slack_conversations.md) table,

>  - messages into the [`slack_conversation_histories`](https://github.com/cloudquery/cloudquery/blob/main/plugins/source/slack/docs/tables/slack_conversation_histories.md) and [slack_conversation_replies](https://github.com/cloudquery/cloudquery/blob/main/plugins/source/slack/docs/tables/slack_conversation_replies.md) tables,

Author: ```suggestion
 - messages into the [`slack_conversation_histories`](https://github.com/cloudquery/cloudquery/blob/main/plugins/source/slack/docs/tables/slack_conversation_histories.md) and [`slack_conversation_replies`](https://github.com/cloudquery/cloudquery/blob/main/plugins/source/slack/docs/tables/slack_conversation_replies.md) tables,
```

 where
     c.is_channel
     and not c.is_archived 
 group by c.name 
 order by messages desc
```

(Note: the CloudQuery plugin only collects data for channels that the bot has been added to.)

Or, perhaps we are curious about the most active users in our workspace. We can write a query that counts the number of messages sent in public channels by each user:

```sql
select 
    u.name, count(h.user) 
from slack_conversation_histories h 
    join slack_conversation_replies r on h.ts = r.conversation_history_ts 
    join slack_users u on u.id = h.user 
group by u.name order by count desc;
```

We can even break this down by day, week, or month to graph how active users are over time:

```sql
select
     u.name,
     date_trunc('day', to_timestamp(round(h.ts::float))) as day,
     count(h.user)
from slack_conversation_histories h
     join slack_conversation_replies r on h.ts = r.conversation_history_ts
     join slack_users u on u.id = h.user 
group by u.name, day 
order by day, count desc
```

## Getting Started

To get started syncing Slack data, see the [Slack source plugin documentation](/docs/plugins/sources/slack/overview) for instructions.

## What's next

> The Slack API comes with some strict [rate limits](https://api.slack.com/docs/rate-limits) that mean syncing messages from channels with a long history can take a long time. We are going to continue expanding the Slack source plugin, adding support for more resources. If you are interested in a specific Slack resource, feel free to raise an [issue on GitHub](https://github.com/cloudquery/cloudquery/issues). Or if you need some help to get started, join us over on [Discord](https://www.cloudquery.io/discord), we'd love to help.

Author: ```suggestion
The Slack API comes with some strict [rate limits](https://api.slack.com/docs/rate-limits) that mean syncing messages from channels with a long history can take a long time. We are [thinking about ways to address this](https://github.com/cloudquery/cloudquery/issues/5809), please upvote or comment on the Github issue if you are interested!

We are also going to continue expanding the Slack source plugin, adding support for more resources as they become available in the Slack API. If you are interested in a specific Slack resource, feel free to raise an [issue on GitHub](https://github.com/cloudquery/cloudquery/issues). Or if you need some help to get started, join us over on [Discord](https://www.cloudquery.io/discord), we'd love to help.
```

---
title: Introducing the Slack Source Plugin
tag: announcement
date: 2022/12/20
description: >-

>   CloudQuery is an open-source, extensible framework that gives you a
>   single-pane-of-glass to your cloud-infrastructure using SQL. Today, we are
>   happy to announce the release of the Slack source plugin.

Author: ```suggestion
  CloudQuery is an open source high performance data integration 
  platform designed for security and infrastructure teams. Today, we are
  happy to announce the release of the Slack source plugin.
```
